src/Database/Log/QueryLogger.php

```diff
@@ -68,7 +68,19 @@ protected function _interpolate($query)
                 return $p ? '1' : '0';
             }
 
-            return is_string($p) ? "'$p'" : $p;
+            if (is_string($p)) {
+                $replacements = [
+                    '$' => '\\$',
+                    '\\' => '\\\\\\\\',
+                    "'" => "''",
+                ];
+
+                $p = strtr($p, $replacements);
```

Comment:
My mistake, I misread the code. You are modifying the `$params` var not `$keys`.